Plugin~/Src/mscore/msCoreAPI.cpp

```diff
@@ -543,13 +543,22 @@ msAPI quatf msParentConstraintGetRotationOffset(const ms::ParentConstraint *self
 msAPI int msSceneGetNumAssets(const ms::Scene *self) { return (int)self->assets.size(); }
 msAPI int msSceneGetNumEntities(const ms::Scene *self) { return (int)self->entities.size(); }
 msAPI int msSceneGetNumConstraints(const ms::Scene *self) { return (int)self->constraints.size(); }
+msAPI int msSceneGetNumInstanceInfos(const ms::Scene* self) { return (int)self->instanceInfos.size();}
 msAPI ms::Asset* msSceneGetAsset(const ms::Scene *self, int i) { return self->assets[i].get(); }
 msAPI ms::Transform* msSceneGetEntity(const ms::Scene *self, int i) { return self->entities[i].get(); }
 msAPI ms::Constraint* msSceneGetConstraint(const ms::Scene *self, int i) { return self->constraints[i].get(); }
+msAPI ms::InstanceInfo* msSceneGetInstanceInfo(const ms::Scene* self, int i) { return self->instanceInfos[i].get(); }
 msAPI bool msSceneSubmeshesHaveUniqueMaterial(const ms::Scene *self) { return self->submeshesHaveUniqueMaterial(); }
 msAPI ms::SceneProfileData msSceneGetProfileData(const ms::Scene *self) { return self->profile_data; }
 #pragma endregion
 
+#pragma region InstanceInfo
+msAPI const char* msInstanceInfoGetPath(const ms::InstanceInfo* self) { return self->path.c_str(); }
+msAPI int msInstanceInfoPropGetArrayLength(const ms::InstanceInfo* self) { return self->transforms.size(); }
+msAPI void msInstanceInfoCopyTransforms(const ms::InstanceInfo* self, float4x4* dst) {
+    memcpy(dst, self->transforms.data(), self->transforms.size() * sizeof(float4x4));
+}
```

Comment:
I would prefer to avoid the copy here, but I am not 100% what happens with the ownership of the array